interface/modules/custom_modules/oe-module-faxsms/library/phone-services/voice_webhook.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+/**
+ * Web Hook for RingCentral Voice Events
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2025 Jerry Padgett <sjpadgett@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+error_log("RingCentral webhook accessed at " . date('Y-m-d H:i:s'));
+
+// Handle RingCentral validation token
+$validationToken = $_SERVER['HTTP_VALIDATION_TOKEN'] ?? '';
+if (!empty($validationToken)) {
+    header("Validation-Token: {$validationToken}");
+    header('Content-Type: text/plain');
+    echo $validationToken;
+    http_response_code(200);
+    error_log("RingCentral validation token returned: {$validationToken}");
+    exit;
+}
+
+// Security: Verify webhook token
+$ignoreAuth = true; // Ignore OpenEMR authentication for this webhook
+require_once(__DIR__ . '/../../../../../globals.php');
+$expectedToken = sqlQuery(
+    "SELECT gl_value FROM globals WHERE gl_name = 'ringcentral_voice_token'"
+)['gl_value'] ?? '';
+
+$providedToken = $_GET['token'] ?? '';
+if (empty($expectedToken) || $providedToken !== $expectedToken) {
```

Comment:
I can see the authentication is done here. :clinking_glasses:

Reply:
Since using this for authentication, would ensure it is an encrypted token.

Reply:
It has a very short life but it seems sloppy where I was just in a hurry to get he feature to work.